crates/fuel-core/src/service/query.rs

```diff
@@ -0,0 +1,88 @@
+//! Queries we can run directly on `FuelService`.
+
+use std::sync::Arc;
+
+use fuel_core_types::{
+    fuel_tx::{
+        Cacheable,
+        Transaction,
+        UniqueIdentifier,
+    },
+    fuel_types::Bytes32,
+    services::txpool::InsertionResult,
+};
+use futures::{
+    Stream,
+    StreamExt,
+};
+
+use crate::{
+    query::transaction_status_change,
+    schema::tx::types::TransactionStatus,
+};
+
+use super::*;
+
+impl FuelService {
+    /// Submit a transaction to the txpool.
+    pub fn submit(&self, mut tx: Transaction) -> anyhow::Result<InsertionResult> {
```

Comment:
👍🏻 Putting the rust-level APIs here will be helpful for migrating to other RPCs and also for those using fuel-core as a library (indexer)